### PR TITLE
PP-4161 Do not show billing address

### PR DESCRIPTION
**Pull request by DanailMinchev**
## WHAT

Do not show billing address on "enter card details" and "confirmation" pages.

## HOW

- Modified `Service.class` to have new `collectBillingAddress` property
- Modified `charge_controller` to ignore and not send `address` property when
  calling connector's `/v1/frontend/charges/{chargeId}/cards` endpoint and
  when `collectBillingAddress` is `false`
- Modified `charge.njk` and `confirm.njk` to not show billing address
  when `service.collectBillingAddress` is `false`
- Modified tests

with @danworth

### CODE REVIEW

File: app/services/normalise_charge.js [hunk 1]
```diff
@@ -1,15 +1,15 @@
 'use strict'
 
-// NPM dependencies
+// npm dependencies
```
Comment (jonheslop) [inline app/services/normalise_charge.js]:
```suggestion
// NPM dependencies
```
Not sure why this changed to lower case?

File: app/services/normalise_charge.js [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
 const humps = require('humps')
-const _ = require('lodash')
+const lodash = require('lodash')
 
-// Local dependencies
+// local dependencies
```
Comment (jonheslop) [inline app/services/normalise_charge.js]:
```suggestion
// Local dependencies
```

Same here

File: app/services/normalise_charge.js [hunk 3]
```diff
@@ -62,8 +62,8 @@ module.exports = (function () {
       'chargeId'
     ]
 
-    _.forIn(body, function (value, key) {
-      if (!_.includes(toIgnore, key)) {
+    lodash.forIn(body, function (value, key) {
```
Comment (jonheslop) [inline app/services/normalise_charge.js]: Nice one for changing this though :)

File: app/services/normalise_charge.js [hunk 4]
```diff
@@ -139,13 +141,13 @@ module.exports = (function () {
...
   return {
-    charge: _charge,
-    addressForApi: addressForApi,
-    addressLines: addressLines,
-    whitespace: whitespace,
-    addressForView: addressForView,
-    creditCard: creditCard,
-    expiryDate: expiryDate,
-    apiPayload: apiPayload
+    charge,
```
Comment (jonheslop) [inline app/services/normalise_charge.js]: 👍

File: test/charge_ui_tests.js [hunk 5]
```diff
@@ -1,9 +1,15 @@
-var path = require('path')
-const _ = require('lodash')
-var renderTemplate = require(path.join(__dirname, '/test_helpers/html_assertions.js')).render
-var cheerio = require('cheerio')
-var should = require('chai').should() // eslint-disable-line
+'use strict'
 
+// npm dependencies
```
Comment (jonheslop) [inline test/charge_ui_tests.js]:
```suggestion
// NPM dependencies
```

File: test/charge_ui_tests.js [hunk 6]
```diff
@@ -1,9 +1,15 @@
...
+const path = require('path')
+const lodash = require('lodash')
+const cheerio = require('cheerio')
+const should = require('chai').should() // eslint-disable-line
+
+// local dependencies
```
Comment (jonheslop) [inline test/charge_ui_tests.js]:
```suggestion
// Local dependencies
```

File: test/fixtures/service_fixtures.js [hunk 7]
```diff
@@ -1,10 +1,10 @@
 'use strict'
 
-// Local dependencies
+// local dependencies
```
Comment (jonheslop) [inline test/fixtures/service_fixtures.js]:
```suggestion
// Local dependencies
```

File: test/fixtures/service_fixtures.js [hunk 8]
```diff
@@ -1,10 +1,10 @@
...
 const pactBase = require('./pact_base')
 const random = require('../../app/utils/random')
 
-// Global setup
+// global setup
```
Comment (jonheslop) [inline test/fixtures/service_fixtures.js]:
```suggestion
// Global setup
```

File: test/integration/charge_billing_address_ft_tests.js [hunk 9]
```diff
@@ -0,0 +1,223 @@
+'use strict'
+
+// npm dependencies
```
Comment (jonheslop) [inline test/integration/charge_billing_address_ft_tests.js]:
```suggestion
// NPM dependencies
```

File: test/integration/charge_billing_address_ft_tests.js [hunk 10]
```diff
@@ -0,0 +1,223 @@
...
+const lodash = require('lodash')
+const nock = require('nock')
+const chai = require('chai')
+const cheerio = require('cheerio')
+const winston = require('winston')
+const expect = chai.expect
+const proxyquire = require('proxyquire')
+const AWSXRay = require('aws-xray-sdk')
+
+// local dependencies
```
Comment (jonheslop) [inline test/integration/charge_billing_address_ft_tests.js]:
```suggestion
// Local dependencies
```

File: test/integration/charge_billing_address_ft_tests.js [hunk 11]
```diff
@@ -0,0 +1,223 @@
...
+const helper = require('../test_helpers/test_helpers')
+const {
+  getChargeRequest,
+  postChargeRequest,
+  defaultConnectorResponseForGetCharge,
+  defaultAdminusersResponseForGetService
+} = helper
+const State = require('../../config/state')
+
+// constants
```
Comment (jonheslop) [inline test/integration/charge_billing_address_ft_tests.js]:
```suggestion
// Constants
```

File: test/integration/charge_ft_tests.js [hunk 12]
```diff
@@ -1,6 +1,6 @@
 'use strict'
 
-// NPM dependencies
+// npm dependencies
```
Comment (jonheslop) [inline test/integration/charge_ft_tests.js]:
```suggestion
// NPM dependencies
```

File: test/integration/charge_ft_tests.js [hunk 13]
```diff
@@ -12,8 +12,22 @@ const proxyquire = require('proxyquire')
 const AWSXRay = require('aws-xray-sdk')
 const should = chai.should()
 
-// Local dependencies
-const app = proxyquire('../../server.js', {
+// local dependencies
```
Comment (jonheslop) [inline test/integration/charge_ft_tests.js]:
```suggestion
// Local dependencies
```

File: test/integration/charge_ft_tests.js [hunk 14]
```diff
@@ -12,8 +12,22 @@ const proxyquire = require('proxyquire')
...
+  postChargeRequest,
+  defaultConnectorResponseForGetCharge,
+  defaultAdminusersResponseForGetService,
+  connectorResponseForPutCharge
+} = helper
+const State = require('../../config/state')
+const serviceFixtures = require('../fixtures/service_fixtures')
+const random = require('../../app/utils/random')
+
+// constants
```
Comment (jonheslop) [inline test/integration/charge_ft_tests.js]:
```suggestion
// Constants
```

File: test/test_helpers/test_helpers.js [hunk 15]
```diff
@@ -1,6 +1,6 @@
 'use strict'
 
-// NPM dependencies
+// npm dependencies
```
Comment (jonheslop) [inline test/test_helpers/test_helpers.js]:
```suggestion
// NPM dependencies
```

**Comment by DanailMinchev** [comment]
Thanks for the review @jonheslop 

I uncapitalised the comments to be consistent with:
- https://github.com/alphagov/pay-frontend/blob/43a00b9eba7fc8fee389c83b5bef6ef60b11c3eb/app/middleware/resolve_service.js#L3
- https://github.com/alphagov/pay-frontend/blob/master/app/utils/charge_validation_fields.js#L6
- https://github.com/alphagov/pay-frontend/blob/master/app/utils/charge_validation_backend.js#L3
- https://github.com/alphagov/pay-frontend/blob/master/app/utils/charge_validation.js#L3
etc

I was not aware about https://eslint.org/docs/rules/capitalized-comments , thanks for pointing this out 👍 

I will update all of the files as discussed (not only files related to the PR)

Thanks

File: app/services/normalise_charge.js (hunk 1 above)
Comment (DanailMinchev) [reply app/services/normalise_charge.js]: I uncapitalised the comments to be consistent with:
- https://github.com/alphagov/pay-frontend/blob/43a00b9eba7fc8fee389c83b5bef6ef60b11c3eb/app/middleware/resolve_service.js#L3
- https://github.com/alphagov/pay-frontend/blob/master/app/utils/charge_validation_fields.js#L6
- https://github.com/alphagov/pay-frontend/blob/master/app/utils/charge_validation_backend.js#L3
- https://github.com/alphagov/pay-frontend/blob/master/app/utils/charge_validation.js#L3
etc

I was not aware about https://eslint.org/docs/rules/capitalized-comments , thanks for pointing this out 👍

**Pull request merged**